### ISSUE TITLE
firstdayofweek should support Sunday beginning weeks

### ISSUE THREAD

**Issue by femtotrader** [JuliaLang/julia#19208]
According
https://www.timeanddate.com/calendar/days/monday.html

Monday is the first day of the week according to the international standard ISO 8601, but in the US, Canada, and Japan it's counted as the second day of the week.

First or second day?
Monday is considered the first day of the week according to the international standard ISO 8601. In many cultures and languages, Monday is given a name that means either “second day” marking Sunday to be the first day of the week or a name that means the day after Sunday.
Slavic languages number Monday as the first day, not the second day. Many European countries have calendars that mostly show Monday as the first day of the week.

So maybe `firstdayofweek` should support a second argument to return Sunday.

See
https://github.com/JuliaLang/julia/blob/fde191ecc9199aa48feb8fed08f9dde7bb699aed/base/dates/adjusters.jl
https://github.com/JuliaLang/julia/blob/1d2913d03af693991b41d3c02d1fc1b99f17811c/test/dates/adjusters.jl

**Comment by nalimilan** [comment]
The .NET interface looks like a good example: https://msdn.microsoft.com/library/microsoft.visualbasic.dateandtime.weekday(v=vs.110).aspx

**Comment by femtotrader** [comment]
Yes... except that second (optional) argument default value should be Monday instead of Sunday to avoid to change default results.

I also wonder if days shouldn't be enum instead of Int.

```
julia> Dates.Monday
1

julia> typeof(Dates.Monday)
Int64
```

**Comment by simonbyrne** [comment]
+1 to both an optional argument (with default as Monday), and making days an enum (say `DayOfWeek`?)

**Comment by JeffreySarnoff** [comment]
The Enum itself should conform to ISO-8601: from 1 (Monday) to 7 (Sunday)

**Comment by simonbyrne** [comment]
Yes, my PR (#19212) does that.
